src/components/App.jsx

```diff
@@ -11,80 +11,117 @@ class App extends React.Component {
 
 		this.state = {
 			text: "",
-			products: products,
+			products,
 			cart: {
 				cartProducts: [],
 				totalPrice: 0,
 			},
 		};
 	}
 
-	changeText = (text) => {
-		this.setState({
-			text: text,
-		});
-	};
-
 	toogleProductToCart = (id) => {
-		this.state.products.map((item) => {
+		const { cart, products } = this.state;
+		products.map((item) => {
+			// I need this "item"...
 			if (item.id === id) {
-				if (this.state.cart.cartProducts.includes(item)) {
+				if (!!cart.cartProducts.find((prod) => prod.id === id)) {
```

Comment:
please figure out if it is possible to leave only one state change and not have if/else